code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/IdentityItem.java

```diff
@@ -42,15 +43,15 @@ public final class IdentityItem {
 	 * @param id                 id for the item; should not be null
 	 * @param authenticatedState {@link AuthenticatedState} for the item; if none is provided {@link AuthenticatedState#AMBIGUOUS} is used as default
 	 * @param primary            primary flag for the item
-	 * @throws IllegalArgumentException if id is null
+	 * @throws IllegalArgumentException if id is null or empty
 	 */
 	public IdentityItem(
 		@NonNull final String id,
 		@Nullable final AuthenticatedState authenticatedState,
 		final boolean primary
 	) {
-		if (id == null) {
-			throw new IllegalArgumentException("id must be non-null");
+		if (StringUtils.isNullOrEmpty(id)) {
+			throw new IllegalArgumentException("id must be non-null and non-empty");
```

Comment:
is there a way to avoid throwing an exception for empty id value? I am thinking this could potentially break an existing implementation which erroneously sends empty ids